src/riak_kv_wm_timeseries_listkeys.erl

```diff
@@ -77,164 +74,122 @@ service_available(RD, Ctx = #ctx{riak = RiakProps}) ->
+    {Result, RD, Ctx}.
 
--spec malformed_request(#wm_reqdata{}, #ctx{}) -> ?CB_RV_SPEC.
-malformed_request(RD, Ctx = #ctx{api_version = "v1"}) ->
-    {false, RD, Ctx};
-malformed_request(RD, Ctx = #ctx{api_version = UnsupportedVersion}) ->
-    handle_error({unsupported_version, UnsupportedVersion}, RD, Ctx).
-
-
--spec allowed_methods(#wm_reqdata{}, #ctx{}) ->
-    {[atom()], #wm_reqdata{}, #ctx{}}.
+-spec allowed_methods(#wm_reqdata{}, #ctx{}) -> cb_rv_spec([atom()]).
 %% @doc Get the list of methods this resource supports.
 allowed_methods(RD, Ctx) ->
     {['GET'], RD, Ctx}.
 
-
--spec resource_exists(#wm_reqdata{}, #ctx{}) ->
-                             {boolean(), #wm_reqdata{}, #ctx{}}.
-resource_exists(RD, #ctx{table = Table} = Ctx) ->
-    Mod = riak_ql_ddl:make_module_name(Table),
-    case catch Mod:get_ddl() of
-        {_, {undef, _}} ->
-            handle_error({no_such_table, Table}, RD, Ctx);
-        _ ->
-            {true, RD, Ctx}
-    end.
-
+-spec resource_exists(#wm_reqdata{}, #ctx{}) -> cb_rv_spec(boolean()).
+resource_exists(RD, #ctx{mod=Mod} = Ctx) ->
+    {riak_kv_wm_ts_util:table_module_exists(Mod), RD, Ctx}.
 
 -spec encodings_provided(#wm_reqdata{}, #ctx{}) ->
-                                {[{Encoding::string(), Producer::function()}],
-                                 #wm_reqdata{}, #ctx{}}.
+                                cb_rv_spec([{Encoding::string(), Producer::function()}]).
 %% @doc List the encodings available for representing this resource.
 %%      "identity" and "gzip" are available.
 encodings_provided(RD, Ctx) ->
     {riak_kv_wm_utils:default_encodings(), RD, Ctx}.
 
 -spec content_types_provided(#wm_reqdata{}, #ctx{}) ->
-                                    {[{ContentType::string(), Producer::atom()}],
-                                     #wm_reqdata{}, #ctx{}}.
+                                    cb_rv_spec([{ContentType::string(), Producer::atom()}]).
 %% @doc List the content types available for representing this resource.
 content_types_provided(RD, Ctx) ->
-    {[{"application/json", produce_doc_body}], RD, Ctx}.
-
+      {[{"text/html", produce_doc_body}], RD, Ctx}.
 
-produce_doc_body(RD, Ctx = #ctx{table = Table,
+produce_doc_body(RD, Ctx = #ctx{table = Table, mod=Mod,
                                 client = Client}) ->
-    F = fun() ->
-                {ok, ReqId} = riak_client:stream_list_keys(
-                                {Table, Table}, undefined, Client),
-                stream_keys(ReqId)
-        end,
-    {{stream, {<<>>, F}}, RD, Ctx}.
-
-stream_keys(ReqId) ->
+    {ok, ReqId} = riak_client:stream_list_keys(
+                    {Table, Table}, undefined, Client),
+    lager:log(info, self(), "in produce_doc_body ~p", [Table]),
+    {{halt, 200}, wrq:set_resp_body({stream, prepare_stream(ReqId, Table, Mod)}, RD), Ctx}.
+
+prepare_stream(ReqId, Table, Mod) ->
+    {<<"<html>">>, fun() -> stream_keys(ReqId, Table, Mod) end}.
```

Comment:
Discussed return format with @jonmeredith - we will encode everything as JSON. Which makes this inconsistency go away.